c10/core/InferenceMode.cpp

```diff
@@ -0,0 +1,15 @@
+
+#include <c10/core/InferenceMode.h>
+
+#include <stdexcept>
+
+namespace c10 {
+
+// Invariant:
+//   Normal mode: InplaceOrView in TLS included
+//   InferenceMode: InferenceMode not in TLS included
```

Comment:
typo s/InferenceMode/InplaceOrView/

Reply:
Also this isn't really an invariant; you're basically defining inference mode to be the state where inplaceorview is not in TLS included